src/v/raft/mux_state_machine.h

```diff
@@ -279,20 +307,17 @@ ss::future<> mux_state_machine<T...>::apply(model::record_batch b) {
           *state);
 
         return result_f.then([this, last_offset](std::error_code ec) {
```

Comment:
Is it guaranteed that `apply_update` will always return an error code and won't throw an exception? I think in case of an exception waiters won't be notified.

Reply:
was this resolved @mmaslankaprv